src/map/script.c

```diff
@@ -7496,6 +7496,86 @@ BUILDIN(getelementofarray)
 	return true;
 }
 
+BUILDIN(array_find)
+{
+	struct script_data *data = script_getdata(st, 2);
+	struct reg_db *ref = NULL;
+	const char *name;
+	int i = 0, id, start_index, array_size;
+
+	if (!data_isreference(data)) {
+		ShowError("buildin_array_find: not a variable\n");
+		script->reportdata(data);
+		script_pushnil(st);
+		st->state = END;
+		return false;
+	}
+
+	start_index = reference_getindex(data);
+	if (start_index < 0 || start_index > SCRIPT_MAX_ARRAYSIZE) {
+		ShowError("buildin_array_find: starting index out of range (%d)\n", start_index);
+		script->reportdata(data);
+		script_pushnil(st);
+		st->state = END;
+		return false;
+	}
+
+	id = reference_getid(data);
+	name = reference_getname(data);
+	ref = reference_getref(data);
+
+	array_size = script->array_highest_key(st, st->rid ? script->rid2sd(st) : NULL, name, ref);
+	if (array_size >= SCRIPT_MAX_ARRAYSIZE) { // edge case
```

Comment:
likewise, this here would never be triggered

Reply:
well when I made this script command, I haven't made [SCRIPT_MAX_ARRAYSIZE PR](https://github.com/HerculesWS/Hercules/pull/2093) yet
it is ACTUALLY this condition made me do that PR

but yeah, if that PR is merge, this condition becomes redundant